ext/opcache/zend_persist.c

```diff
@@ -903,10 +903,11 @@ zend_class_entry *zend_persist_class_entry(zend_class_entry *orig_ce)
 			ce->default_static_members_table = zend_shared_memdup_free(ce->default_static_members_table, sizeof(zval) * ce->default_static_members_count);
 
 			/* Persist only static properties in this class.
-			 * Static properties from parent classes will be handled in class_copy_ctor */
-			i = (ce->parent && (ce->ce_flags & ZEND_ACC_LINKED)) ? ce->parent->default_static_members_count : 0;
-			for (; i < ce->default_static_members_count; i++) {
-				zend_persist_zval(&ce->default_static_members_table[i]);
+			 * Static properties from parent classes will be handled in class_copy_ctor and are marked with IS_INDIRECT */
+			for (i = 0; i < ce->default_static_members_count; i++) {
+				if (Z_TYPE(ce->default_static_members_table[i]) != IS_INDIRECT) {
```

Comment:
If `i > ce->parent->default_static_members_count` then is this check unnecessary?

But this check doesn't seem that expensive, so it might not be worth changing the code....

Reply:
That would be a correct change, but I don't expect it to be worthwhile. In the cases where we can skip, we check the type too in `zend_persist_zval`, so it's not like we can avoid to fetch the value into the cache; and I expect this to be reasonably cheap.
More interesting would be a way to avoid the check (or avoid a big part of the loop) if we are guaranteed to _not_ needing to persist the value.

Reply:
I thought could skip it if `ce->parent->num_traits == 0`, but I'm not sure if this is actually possible.

Reply:
@nielsdos 

Just to confirm: Is this idea actually useless?

Reply:
I'm unsure. I'd need to think of possible consequences in edge cases. For example, a class may inherit from another one that has a trait, maybe even several inheritance leveks deep.

Reply:
I see. I thought this was a recursive process. So I thought that static members from parent classes would always be added to persistence already, unless the direct parent used traits.

But I'm not very knowledgeable here so I'm not sure. Thanks for confirmation!